src/Markdig/Extensions/Bootstrap/BootstrapAlertRenderer.cs

```diff
@@ -0,0 +1,72 @@
+// Copyright (c) Alexandre Mutel. All rights reserved.
+// This file is licensed under the BSD-Clause 2 license.
+// See the license.txt file in the project root for more information.
+
+using System.Linq;
+using Markdig.Extensions.Alerts;
+using Markdig.Renderers;
+using Markdig.Renderers.Html;
+using Markdig.Syntax;
+
+namespace Markdig.Extensions.Bootstrap;
+
+/// <summary>
+/// A HTML renderer for a <see cref="AlertBlock"/> that adds bootstrap classes.
+/// </summary>
+/// <seealso cref="HtmlObjectRenderer{AlertBlock}" />
+public class BootstrapAlertRenderer : HtmlObjectRenderer<AlertBlock>

```

Comment:
I would think that you would not need to create a new `BootstrapAlertRenderer` class.

You can add the attributes as we do directly in the boostrap extension and then you can add if not already the existing [AlertBlockRenderer](https://github.com/xoofx/markdig/blob/master/src/Markdig/Extensions/Alerts/AlertBlockRenderer.cs) and set the RenderKind property to discard the rendering of the SVG. In the end it should be equivalent to the code you wrote with a bit less duplication.

Also, in general, a renderer should not modify the model that it renders (so AddProperty should not be done there)

Reply:
Thinking about it, I would not remove the Note text with the SVG from the Alert block when activating Bootstrap. We should probably only keep bootstrap extension to add the attributes.

Reply:
> Thinking about it, I would not remove the Note text with the SVG from the Alert block when activating Bootstrap. [...]

I disagree for multiple reasons:
- I believe that this library should not modify WHAT is rendered (the user determines that by writing markdown), but only HOW the content is rendered. Here we are already using bootstrap (at the users choice), which makes an effort to highlight this visually already. Therefore we don't need extra text as we do without bootstrap.
- That text has nothing to distinguish it from the content. It looks like part of it rather than some sort of headline that was added automatically.
- The icons are quite hard to make out when using the bootstrap dark theme.
- Not utilizing boostrap-icons but additional SVGs. However, we should not blindly use boostrap-icons either because that is a separate CSS/font that is not shipped with bootstrap-core by default.
- There is literally no space between the icon and the text, which makes it look crowded.

![image](https://github.com/xoofx/markdig/assets/6608231/c82c0c78-6477-4382-8c5d-71e8701c19d3)

Reply:
Fair enough